### ISSUE TITLE
Xamarin iOS gesture recognizers do not work in 17.0.1

### ISSUE THREAD

**Issue by vsfeedback** [xamarin/Xamarin.Forms#14956]
_This issue has been moved from [a ticket on Developer Community](https://developercommunity.visualstudio.com/t/Xamarin-iOS-gesture-recognizers-do-not-w/1586602)._

---
[regression] [worked-in:VS 2019]
Exactly as the title says. The app deploys and runs, however any controls using gesture recognizers do not respond to taps.
I am deploying to a USB-connected iPhone. It works fine on Android.

Some example code:

<Frame BorderColor="Transparent" HasShadow="False" Padding="20,10,20,10" HorizontalOptions="Start">
                        <Frame.GestureRecognizers>
                            <TapGestureRecognizer Tapped="OnBackTapped" NumberOfTapsRequired="1" />
                        </Frame.GestureRecognizers>
                        <Image Source="back24.png"></Image>
                    </Frame>

---
#### Feedback Bot on 11/17/2021, 00:01 AM: 

<p>We have directed your feedback to the appropriate engineering team for further evaluation. The team will review the feedback and notify you about the next steps.</p>

#### Graeme Vincent on 11/17/2021, 04:50 AM: 

<p>I think I need to point out that the gesture recognizers do work when I deploy to an iPhone connected to a Macbook.<br>
That deployment has it’s own new set of problems which I have logged in another ticket.</p>

#### Luke Sullivan on 11/18/2021, 09:43 PM: 

<p>I’m having the same problem.  This is a major hinderance to our development schedule.  Any workaround or solutions would be great!  Thank you!</p>

#### Luke Sullivan on 11/19/2021, 10:45 AM: 

<p>I would also like to point out that any iOS 15 device stopped deploying on vs 2019 as well.  The workout fix was to take files from vs 2022 to allow vs2019 to then be able to deploy again.  This solved that issue but now the gesture recognition doesn’t work in vs 2019 16.11.7 either.  This is not restricted to vs 2022 v17.0.1</p>

---
### Original Solutions
(no solutions)

**Comment by jfversluis** [comment]
Can't reproduce both on the Simulator as well as on a physical device. Please ping me when you find this and can add a reproduction project for us to try.

**Comment by lukies97** [comment]
I don't know of a project you can try, but I am still having this issue.  It does not happen when using the simulator, only when loading directly to a phone.  In Android the gesture recognizer works, but in iOS it does not work.  This has not been fixed.

**Comment by melmulcahy32** [comment]
I was able to reproduce in 2022 Community using the [WorkingWithGestures Sample Project](https://github.com/xamarin/xamarin-forms-samples/tree/main/WorkingWithGestures/TapGesture) on a USB connected iPhone XS.

**Comment by jfversluis** [comment]
@melmulcahy32 thanks! Not sure what to tell you... For me that just works... 😩

**Comment by melmulcahy32** [comment]
> @melmulcahy32 thanks! Not sure what to tell you... For me that just works... 😩

😭 perhaps it's an environment thing. I know my dev team uses MacStadium and builds apps using a virtual Mac.

**Comment by grvingit** [comment]
I was the user who reported this. I think it may be worth mentioning that I followed some other steps from a Microsoft rep to fix a provisioning problem in VS2019. This was to install VS 2022, then copy the iSign.Core.dll, iSign.Core.pdb and System.Formats.Asn1.dll files from /tools/msbuild/ios to C:\Program Files (x86)\Microsoft Visual Studio\2019\Enterprise\MSBuild\Xamarin\iOS.
Unfortunately I cannot remember if the gesture recognizers bug appeared before or after copying those files. I'm pretty sure it was after, because I wasn't able to deploy at all until copying those files.

**Comment by jfversluis** [comment]
@grvingit but you are still seeing this behavior?

**Comment by MiekVenter** [comment]
I have the same issue, with provided WorkingWithGestures sample and my own codes. Android no issues on same apps. . USB direct with hot reload, no MAC pairing, studio 2019 and 2022.

**Comment by lukies97** [comment]
I also followed the same steps as @grvingit and I am still having trouble with the gesture recognizer.  I didn't have the issues in 2019 until after followed those steps and now I have the issue in VS2019 and VS2022.

**Comment by grvingit** [comment]
@jfversluis Yes, I am still seeing it. I had to remove all gesture recognizers in my app and replace them with image buttons in order to get an urgent update out, but this evening when I tried them again they still do not work.

**Comment by jfversluis** [comment]
@MiekVenter did you also do something with the copying of DLLs as mentioned above?

**Comment by MiekVenter** [comment]
No, no copying. New clean 2022 installation, first project ever compiled on new installation was the demo.

**Comment by jfversluis** [comment]
You're running this from Windows through Hot Restart (directly on a physical iOS device)? Or do you have it paired to a Mac?

**Comment by MiekVenter** [comment]
Direct on hot restart

**Comment by MiekVenter** [comment]
I will have an IOs box by next week.

**Comment by jfversluis** [comment]
@chamons @dalexsoto I see this happened before in #10162 any chance this has regressed?

**Comment by grvingit** [comment]
For me the problem only occurs with Hot Restart. I've just tried it through a paired Mac and gesture recognizers works fine. Unfortunately for me developing and debugging this way is not an option as deploying is too slow - it takes a number of minutes  each time.

**Comment by jfversluis** [comment]
Yeah the cause seems to be Hot Restart here, found an earlier issue that describes the same, maybe there is a regression there. We just need to wait for them to wake up ;)

**Comment by lukies97** [comment]
Did they ever wake up?  ;)  10 Days is a long sleep.

**Comment by jfversluis** [comment]
They work very hard! No news yet unfortunately :(

**Comment by lukies97** [comment]
I appreciate that they work hard.  I was just trying at some levity.  Hopefully they can figure out something soon.

**Comment by lukies97** [comment]
I appreciate that they work hard.  I was just trying at some levity.  Hopefully they can figure out something soon.

**Comment by BlueRaja** [comment]
Just want to verify that, using Hot Restart, all `GestureRecognizer`s are completely broken, for all controls.  Along with [many](https://github.com/xamarin/xamarin-macios/issues/14013) [many](https://developercommunity.visualstudio.com/t/VS-2022-breaking-on-caught-exception-ev/1650854?from=email&space=8&scope=follow) [other](https://developercommunity.visualstudio.com/t/missing-method-bug-fixed-in-1703-has-regressed-in/1651598?from=email) [critical](https://developercommunity.visualstudio.com/t/Hot-Restart-with-Asset-Catalogs/1177657?from=email&space=8&scope=follow) [bugs](https://developercommunity.visualstudio.com/t/Xamarin-iOS:-Changes-in-app-not-reflecti/1586652?from=email&space=8&scope=follow).

Really not a good experience so far.

**Comment by jfversluis** [comment]
I've raised it again to the people responsible, thanks for your patience!

**Comment by melmulcahy32** [comment]
I think VS 2022 Preview 5 has a lot of patches regard this issue. It's been promising for my team so far.

**Comment by haavamoa** [comment]
Me and a colleague are now experiencing this on Mac with VS 4 Mac 8.10.17 and 8.10.18. Running M1 MacBooks.

**Comment by haavamoa** [comment]
We are now experiencing this in our production app by using Azure Devops pipeline, because pipeline is using Visual Studio 4 Mac 8.10.18 with the macOS11 virtual environment. Locally on my machine I am running 8.10.19 and it works as expected. 

So if you are experiencing this issue, check your Mac's VS4Mac version.

https://github.com/actions/virtual-environments/issues/5120

**Comment by nemekzg** [comment]
Is there some progress on the issue? We've been affected for almost three months now. Working on UI controls is hardly possible.

**Comment by grvingit** [comment]
I'm the user that reported this. It happens in VS 2019 and VS 2022. It started around the time 2022 came out. It occurs using Hot Restart, but not when the iPhone is connected via a Mac (I'm developing in Windows).
I've had to remove all gesture recognizers from my app, fortunately I was able to. But it does restrict what I want to do.

**Comment by nemekzg** [comment]
@jfversluis: the dump of my current configuration is pasted below. For the last couple of months, I've tried both VS 2019 and VS 2022 (the latter also in the preview version) but the problem is still there. We use Xamarin Forms 5.0.0.2337 and we do use HotRestart. Our setup consists of a Windows machine that deploys the project to an iPhone via USB. We also use a Macincloud instance. The problem seems to disappear at times with no apparent reason and then it's back. I've reinstalled my VS 2022 and this did not help either.

A pecularity of our setup is that we have downgraded System.Memory first to 4.5.2 (due to problems with SQLite - https://github.com/praeclarum/sqlite-net/issues/1074) and then to 4.5.0 (due to problems with SignalR - https://github.com/xamarin/Xamarin.Forms/issues/15311) but gesture recognizer had not worked before.


Microsoft Visual Studio Enterprise 2022
Version 17.1.6
VisualStudio.17.Release/17.1.6+32421.90
Microsoft .NET Framework
Version 4.8.04084

Installed Version: Enterprise

.NET Core Debugging with WSL   1.0
.NET Core Debugging with WSL

ADL Tools Service Provider   1.0
This package contains services used by Data Lake tools

ASA Service Provider   1.0

ASP.NET and Web Tools 2019   17.1.363.30963
ASP.NET and Web Tools 2019

ASP.NET Web Frameworks and Tools 2019   17.1.363.30963
For additional information, visit https://www.asp.net/

Azure App Service Tools v3.0.0   17.1.363.30963
Azure App Service Tools v3.0.0

Azure Data Lake Tools for Visual Studio   2.6.5000.0
Microsoft Azure Data Lake Tools for Visual Studio

Azure Functions and Web Jobs Tools   17.1.363.30963
Azure Functions and Web Jobs Tools

Azure Stream Analytics Tools for Visual Studio   2.6.5000.0
Microsoft Azure Stream Analytics Tools for Visual Studio

C# Tools   4.1.0-5.22165.10+e555772db77ca828b02b4bd547c318387f11d01f
C# components used in the IDE. Depending on your project type and settings, a different version of the compiler may be used.

Common Azure Tools   1.10
Provides common services for use by Azure Mobile Services and Microsoft Azure Tools.

Extensibility Message Bus   1.2.6 (master@34d6af2)
Provides common messaging-based MEF services for loosely coupled Visual Studio extension components communication and integration.

Fabric.DiagnosticEvents   1.0
Fabric Diagnostic Events

Microsoft Azure Hive Query Language Service   2.6.5000.0
Language service for Hive query

Microsoft Azure Service Fabric Tools for Visual Studio   17.0
Microsoft Azure Service Fabric Tools for Visual Studio

Microsoft Azure Stream Analytics Language Service   2.6.5000.0
Language service for Azure Stream Analytics

Microsoft Azure Tools for Visual Studio   2.9
Support for Azure Cloud Services projects

Microsoft JVM Debugger   1.0
Provides support for connecting the Visual Studio debugger to JDWP compatible Java Virtual Machines

Microsoft Library Manager   2.1.161+abc97ecc7d.RR
Install client-side libraries easily to any web project

Microsoft MI-Based Debugger   1.0
Provides support for connecting Visual Studio to MI compatible debuggers

Microsoft Visual Studio Tools for Containers   1.2
Develop, run, validate your ASP.NET Core applications in the target environment. F5 your application directly into a container with debugging, or CTRL + F5 to edit & refresh your app without having to rebuild the container.

Mono Debugging for Visual Studio   17.1.10 (65f98f5)
Support for debugging Mono processes with Visual Studio.

NuGet Package Manager   6.1.0
NuGet Package Manager in Visual Studio. For more information about NuGet, visit https://docs.nuget.org/

Razor (ASP.NET Core)   17.0.0.2206201+62a2c1d6162f828801565a7ca26d9d48b810a05b
Provides languages services for ASP.NET Core Razor.

Snapshot Debugging Extension   1.0
Snapshot Debugging Visual Studio Extension Detailed Info

SQL Server Data Tools   17.0.62203.25080
Microsoft SQL Server Data Tools

ToolWindowHostedEditor   1.0
Hosting json editor into a tool window

TypeScript Tools   17.0.1229.2001
TypeScript Tools for Microsoft Visual Studio

Visual Basic Tools   4.1.0-5.22165.10+e555772db77ca828b02b4bd547c318387f11d01f
Visual Basic components used in the IDE. Depending on your project type and settings, a different version of the compiler may be used.

Visual F# Tools   17.1.0-beta.21610.4+07b5673e4f2fa7630e78abe37f16b372353a7242
Microsoft Visual F# Tools

Visual Studio Code Debug Adapter Host Package   1.0
Interop layer for hosting Visual Studio Code debug adapters in Visual Studio

Visual Studio Container Tools Extensions   1.0
View, manage, and diagnose containers within Visual Studio.

Visual Studio IntelliCode   2.2
AI-assisted development for Visual Studio.

Visual Studio Tools for Containers   1.0
Visual Studio Tools for Containers

VisualStudio.DeviceLog   1.0
Information about my package

VisualStudio.Foo   1.0
Information about my package

VisualStudio.Mac   1.0
Mac Extension for Visual Studio

Xamarin   17.1.0.329 (d17-1@bc74cf0)
Visual Studio extension to enable development for Xamarin.iOS and Xamarin.Android.

Xamarin Designer   17.1.0.87 (remotes/origin/main@d50fb5635)
Visual Studio extension to enable Xamarin Designer tools in Visual Studio.

Xamarin Templates   17.1.8 (590388a)
Templates for building iOS, Android, and Windows apps with Xamarin and Xamarin.Forms.

Xamarin.Android SDK   12.2.4.3 (d17-1/fc1125e)
Xamarin.Android Reference Assemblies and MSBuild support.
    Mono: adf1bc4
    Java.Interop: xamarin/java.interop/d17-1@7ac33610
    ProGuard: Guardsquare/proguard/v7.0.1@912d149
    SQLite: xamarin/sqlite/3.37.1@0893b3b
    Xamarin.Android Tools: xamarin/xamarin-android-tools/d17-1@db125a7


Xamarin.iOS and Xamarin.Mac SDK   15.8.0.5 (cc19b50de)
Xamarin.iOS and Xamarin.Mac Reference Assemblies and MSBuild support.

**Comment by nemekzg** [comment]
@jfversluis 17.2 seems to work fine. We'll see if it stops after some deployments.

**Comment by bobbydharrell** [comment]
I am on a Macbook pro (intel) Visual Studio for Mac 17.6.7, Also tried Using VS 2022 on Virt Windows, this problem occures when in development, and also in production app, but only for the Community Toolkit EventTo Command behavior on our radio buttons. Everything else seems to be fine
